src/development/troubleshoot.md

```diff
@@ -161,3 +135,75 @@ where `<project_id>` is the random-character ID of the project.  That can be fou
 If you see a bare "File not found" error when accessing your Drupal site with a browser, this means that you've pushed your code as a vanilla project but no *index.php* has been found.
 
 Make sure your repository contains an *index.php* file in the [web location root](/configuration/app-containers.md#locations), or that your [Drush](/frameworks/drupal7/drush.md) make files are properly named.
+
+
+## Stuck build (or deployment)
+
+If you see a build or deployment running longer than expected, that may be one of the following cases:
+
+1. The build is blocked by a process in your build hook.
+2. The deployment is blocked by a long running process in your deploy hook.
+3. The deployment is blocked by a long running cron job in the environment.
+4. The deployment is blocked by a long running cron job in the parent environment.
+
+To determine if your environment is being stuck in the build or the deployment, you should look at the build log available on the UI. When you see a log line like below, your application is built and the deployment is stuck:
```

Comment:
Minor nitpick: Remove "you should".  Either change to "you can" or omit entirely.  "You should" feels too much like an accusatory command here.  (Entirely subjective, I know.)